Add OpenNebula one_image module

ilicmilan opened the pull request:
##### SUMMARY
Add `one_image` module to manage images on the [OpenNebula](https://opennebula.org/) open source cloud platform.
While we were developing this module we were inspired by Google Cloud and Amazon's EC2 modules, but we also aligned with how OpenNebula works.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
* one_image

##### ADDITIONAL INFORMATION
We've been using this module in the production for the last 3 months.
We also have integration tests, but they require a running OpenNebula instance, so we will have to discuss how to integrate that with your testing system.

ilicmilan committed:
Add OpenNebula one_image module

`one_image` - module for managing images on OpenNebula

Add integration tests

resmo commented on lib/ansible/modules/cloud/opennebula/one_image.py:
```diff
@@ -0,0 +1,519 @@
+    if regexp and action != 'list':
+        module.fail_json(msg="Parameter 'action' has to be 'list' when the option 'regexp' is specified.")
+
+    if action and action == 'list':
```
a list action or state is something we have accepted in the past, but not anymore for new modules. please create a separate facts module for this functionality.

resmo reviewed (changes requested):
see comment about list functionality inline

resmo reviewed (approved):
As a suggestion, I would skip `action` and move these items to `state`. rename is already idempotent so a  `state=renamed` makes sense to me. `state=clone` would be the only non idempotent "action".

ilicmilan commented:
@resmo `state=renamed` would be also non idempotent.
```
- one_image:
  name: app
  state: renamed
  new_name: app-new

- one_image:
  name: app
  state: renamed
  new_name: app-new
```
What would you expect here?

ilicmilan committed:
Remove the action option, Update tests

resmo commented:
I would implement in a way to look for an image named "app-new" (as the identifier) and if this exists and the image "app" does not exist (anymore), I  would assume the image were already renamed, so `changed=false`.

As a user I would expect that the first task renames the image, the second task would be a no-op `changed=false`. 

I am not sure if this makes sense, any thoughts?

ilicmilan commented:
I've asked few more Ansible users for an opinion and they expect that second task will fail, because the image "app" doesn't exist anymore since it's renamed.
For now, I have made it possible to rename the image only with an `id`.

resmo commented:
fine for me.

resmo reviewed (approved):
counting 4 community shipits + from me
shipit
